h/api/test/schemas_test.py

```diff
@@ -42,13 +42,13 @@ def test_it_sets_appropriate_error_message_when_data_invalid(self):
 class TestAnnotationSchema(object):
```

Comment:
Yeah they're very repetitive, it does look like a strong candidate for a `parametrize()`, definitely.

I _do_ think the tests add value, though. Because I think it's totally possible to create a json schema that I think does one thing, but actually doesn't do exactly what I think, and these tests make sure that it does. The json schema is a complex and very nested data structure that it's totally easy (and I did) make mistakes with. It's either test everything manually or write a set of unit tests.

Also when making changes to a jsonschema, the corresponding unit test changes in the same commit will document (probably better than the commit message) exactly what the programmer was trying to change.

These tests would also allow us to substitute jsonschema with something else (that had the same interface) and know we hadn't changed behaviour.

In what ways are they too coupled to details of jsonschema? Other than the error message strings, but I don't think there's much to be done about that, I do want to test that when the validation is giving an error it's the error that I think it's giving. Otherwise the validation when called by the test could be failing for some completely different reason (e.g. there is something else invalid about the test data), the actual thing I'm trying to test could be broken, and the test could still be passing.

Also I just want to test that we're giving out the right error messages.

I don't think there's any other way to test "jsonschema validation failed for the reason I expected it to fail" other than asserting against the error message strings?